app/models/piece.rb

```diff
@@ -22,11 +22,15 @@ def capture_piece(x, y)
   def valid_move?(destination_x, destination_y)
     valid = true
     game = self.get_game
-    # Check if piece is obstructed
-    if self.is_obstructed?(destination_x, destination_y)
-      valid = false
-      # Check if destination is occupied by a piece of the same color
-    elsif game.is_piece_present?(destination_x, destination_y)
+    #Check to see if piece is not a knight
+    if self.type != "Knight"
```

Comment:
Piece should be naive of Knight. It would be better to add an argument definining whether the piece is allowed to be obstructred into the method with a default to false